sale_product_pack/models/product_pack_line.py

```diff
@@ -24,10 +24,11 @@ def get_sale_order_line_vals(self, line, order):
             "pack_modifiable": line.product_id.pack_modifiable,
         }
         sol = line.new(line_vals)
-        sol.product_id_change()
+        sol._onchange_product_id_warning()
         sol.product_uom_qty = quantity
-        sol.product_uom_change()
-        sol._onchange_discount()
+        sol._compute_product_uom_qty()
+        sol._compute_product_uom()
+        sol._compute_discount()
```

Comment:
Is the calling of these functions really needed?

Why is the function **_onchange_product_id_warning** needed to be called?

**_compute_product_uom_qty** is strange to call it because you set **product_uom_qty**

**_compute_uom** does not use **product_uom_qty** in its calculation

**_compute_discount** depends on **product_uom_qty**

If I remove them all, the test scenario's succeed.
So if they are needed I suppose a test scenario should be created.

Reply:
To be honest, i picked up the changes from July after she closed her pr. I didn't think further. I will check that

Reply:
Hi @rven, 
You are right the _compute are not needed. 
The _onchange_product_id_warning was previously inside the product_id_change. To avoid losing this functionality, I think we should call the onchange method.